api/wallet_test.go

```diff
@@ -1474,3 +1474,158 @@ func TestWalletGETDust(t *testing.T) {
 		t.Fatal("dustThreshold mismatch")
 	}
 }
+
+// testWalletTransactionEndpoint is a subtest that queries the transaction endpoint of a node.
+func testWalletTransactionEndpoint(t *testing.T, st *serverTester) {
+	// Mining blocks should have created transactions for the wallet containing
+	// miner payouts. Get the list of transactions.
+	var wtg WalletTransactionsGET
+	err := st.getAPI("/wallet/transactions?startheight=0&endheight=-1", &wtg)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Query the details of all transactions using
+	// /wallet/transaction/:id
+	for _, txn := range wtg.ConfirmedTransactions {
+		var wtgid WalletTransactionGETid
+		wtgidQuery := fmt.Sprintf("/wallet/transaction/%s", txn.TransactionID)
+		err = st.getAPI(wtgidQuery, &wtgid)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if wtgid.Transaction.TransactionID != txn.TransactionID {
+			t.Fatalf("Expected txn with id %v but was %v", txn.TransactionID, wtgid.Transaction.TransactionID)
+		}
+	}
+}
+
+// testWalletTransactionEndpoint is a subtest that queries the transactions endpoint of a node.
+func testWalletTransactionsEndpoint(t *testing.T, st *serverTester) {
+	var wtg WalletTransactionsGET
+	err := st.getAPI("/wallet/transactions?startheight=0&endheight=-1", &wtg)
+	if err != nil {
+		t.Fatal(err)
+	}
+	totalTxns := len(wtg.ConfirmedTransactions)
```

Comment:
Same here, should have an input passed in that tells you what `totalTxns` is supposed to be, to make sure that you got them all.